pkg/deploy/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -177,10 +209,26 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 	}
 	created, err := c.rn.ReplicationControllers(config.Namespace).Create(deployment)
 	if err != nil {
-		// If the deployment was already created, just move on. The cache could be
-		// stale, or another process could have already handled this update.
+		// We need to find out if our controller owns that deployment and report error if not
 		if kapierrors.IsAlreadyExists(err) {
-			return c.updateStatus(config, existingDeployments)
+			rc, err := c.rcLister.ReplicationControllers(deployment.Namespace).Get(deployment.Name)
+			if err != nil {
+				return fmt.Errorf("error while deploymentConfigController getting the replication controller %s/%s: %v", rc.Namespace, rc.Name, err)
+			}
+			// We need to make sure we own that RC or adopt it if possible
+			isOurs, err := cm.ClaimReplicationController(rc)
+			if err != nil {
+				return fmt.Errorf("error while deploymentConfigController claiming the replication controller: %v", err)
+			}
+			if isOurs {
+				// If the deployment was already created, just move on. The cache could be
+				// stale, or another process could have already handled this update.
+				return c.updateStatus(config, existingDeployments)
+			} else {
```

Comment:
I know, I have changed it 2 times while writing it but somehow I wanted to explicitly state that this is for the case where it isn't ours :) and to be resilient to future changes